File: 1157.c
//Updated code
#include <stdio.h>
int main(){
    char str[1000001];
    int alpha_c[26]={0};
    scanf("%s",str);
    for (int i=0; str[i]!=0; i++){
        if (str[i]>96) alpha_c[str[i]-97]++;
        else alpha_c[str[i]-65]++;
    }
    int mx=0;
    for (int i=0; i<26; i++){
        if (alpha_c[i]>mx) mx=alpha_c[i];
    }
    int has_mx=0;
    int loc=0;
    for (int i=0; i<26; i++){
        if (mx==alpha_c[i]){
            if (has_mx) {
                printf("?");
                return 0;
            }
            has_mx=1;
            loc=i;
        }
    }
    printf("%c",loc+65);
    return 0;
}

File: 1157.py
alpha_c=[0 for _ in range(26)]
s=input()
for alpha in s:
    temp=ord(alpha)
    if temp>96: alpha_c[temp-97]+=1
    else: alpha_c[temp-65]+=1
mx=max(alpha_c); loc=0
has_mx=False
for i in range(26):
    if mx==alpha_c[i]: 
        if has_mx: print("?"); exit(0)
        has_mx=True
        loc=i
print(chr(loc+65))